src/providers/recentmessages/Api.cpp

```diff
@@ -20,72 +20,77 @@ using namespace recentmessages::detail;
 
 void load(const QString &channelName, std::weak_ptr<Channel> channelPtr,
           ResultCallback onLoaded, ErrorCallback onError, const int limit,
-          const long after, const long before)
+          const long after, const long before, const bool jitter)
 {
     qCDebug(LOG) << "Loading recent messages for" << channelName;
 
     const auto url =
         constructRecentMessagesUrl(channelName, limit, after, before);
 
-    NetworkRequest(url)
-        .onSuccess([channelPtr, onLoaded](const auto &result) {
-            auto shared = channelPtr.lock();
-            if (!shared)
-            {
-                return;
-            }
-
-            qCDebug(LOG) << "Successfully loaded recent messages for"
-                         << shared->getName();
-
-            auto root = result.parseJson();
-            auto parsedMessages = parseRecentMessages(root);
-
-            // build the Communi messages into chatterino messages
-            auto builtMessages =
-                buildRecentMessages(parsedMessages, shared.get());
-
-            postToThread([shared = std::move(shared), root = std::move(root),
-                          messages = std::move(builtMessages),
-                          onLoaded]() mutable {
-                // Notify user about a possible gap in logs if it returned some messages
-                // but isn't currently joined to a channel
-                const auto errorCode = root.value("error_code").toString();
-                if (!errorCode.isEmpty())
+    const long delay = jitter ? std::rand() % 100 : 0;
```

Comment:
I would add the unit of this delay to the variable name. You have to first follow the code and then get the documentation of `QTimer::singleShot` to find out they are milliseconds.